src/mrack/providers/aws.py

```diff
@@ -209,9 +209,47 @@ async def validate_hosts(self, reqs):
         """Validate that host requirements are well specified."""
         return
 
+    def get_subnet_available_ips(self, subnet_id, log_msg_start):
+        """Get number of IPs available in a subnet."""
+        try:
+            subnet = self.ec2.Subnet(subnet_id)
+        except ClientError:
+            err_msg = (
+                f"{log_msg_start} Error retrieving " f"info from subnet: {subnet_id}"
+            )
+            logger.error(err_msg)
+            raise
+
+        return subnet.available_ip_address_count
+
     async def can_provision(self, hosts):
-        """Check that hosts can be provisioned."""
-        return True
+        """Check that all host can be provisioned.
+
+        Checks:
+        * Available IPv4 addresses are enough
+        """
+        log_msg_start = self.dsp_name
+
+        available_ips = 0
+        # Get subnets ids from first subnet (they're the same for every host)
+        if subnet_ids := hosts[0].get("subnet_ids"):
+            for subnet_id in subnet_ids:
+                available_ips += self.get_subnet_available_ips(subnet_id, log_msg_start)
+        elif subnet_id := hosts[0].get("subnet_id"):
+            available_ips += self.get_subnet_available_ips(subnet_id, log_msg_start)
+        else:
+            logger.warning(
+                f"{log_msg_start} No subnet/s specified. "
+                "Could not check IP addresses availability."
+            )
+            return True
+
+        logger.info(
+            f"{log_msg_start} There are "
+            f"{available_ips} IPs available and {len(hosts)} hosts to be provisioned"
```

Comment:
Personal note: Tiny detail, I would prefer printing out availability per network in cycle.

Reply:
My idea was having a pool of IPs from the specified subnets.
Example:
- We want to provision 100 host.
- SubnetA has 80 addresses and SubnetB has 50 addresses.
- We start provisioning in SubnetA. Once the are no addresses left in SubnetA, we continue provisioning the remaining 20 hosts in SubnetB. This way we have more capacity than picking a subnet at the beginning of the provisioning. It'd OK to print per Network as additional info, but what matter in this implementation is the global count.

Reply:
If you create something like openstack has:
`python
self.ip_availabilities = {
     "subnet_id_1" = 20,
     "subnet_id_2" = 80,
}
 here in can provision 
you can later reuse it and do not query the pools again that was the point i wanted to point out in our meeting and at the end of this function maybe not in cycle as i initially said you could print nicely this dictionary in similar style we have in openstack already.

Reply:
Comments addressed!
I used something similar to your suggestion